### ISSUE TITLE
BeanMapper performance is poor

### ISSUE THREAD

**Issue by kkrgwbj** [jdbi/jdbi#495]
@stevenschlansker , I use testng tests jdbi2.75, found both in multi threads operation and single thread operation, the performance of this : JDBC > sql2o > mybatis > Jdbi. Jdbi performance is much worse than mybatis. I, about 6000 data in the database query returns a List object, made at least 20 rounds of testing, run 1000 times, respectively, multithreading running and single thread. Are the results. Slower than mybatis probably around 10 s. Mybatis is slower than the sql2o probably less than 10 s,sql2o is slower than jdbc less then 2s.If you need, I can upload my test code

**Comment by stevenschlansker** [comment]
You should consider using a profiler such as [VisualVM](http://visualvm.java.net/) to identify where the code is spending its time.  While it's entirely possible there is a bug in jdbi, we need a hint to go off of -- a profile showing some of our code taking long than it should would be very helpful.  Or if you can upload a small self contained example that is even better.

**Comment by kkrgwbj** [comment]
@stevenschlansker ，I upload the test cases, the attachment is a complete project, you can download down, there are my test SQL, and data, including complete code.

**Comment by stevenschlansker** [comment]
Let's continue this discussion on the mailing list.

**Comment by kkrgwbj** [comment]
@stevenschlansker ，Ok, waiting for your information. Have the test project email sent to you.

**Comment by kkrgwbj** [comment]
@stevenschlansker ,Thank you, I will refer to your jdbi3 reconstruct the next. But the string conversion lowercase is jdbi2.x own actually, I just added a  camel convert.

**Comment by stevenschlansker** [comment]
Yes, that's why I filed the new issue #496

**Comment by kkrgwbj** [comment]
@stevenschlansker ,i test jdbi3,the performance is still ： sql2o>mybatis>jdbi3.Jdbi3 underline convert the Camel, not a lot of toLowerCase ().

**Comment by miere** [comment]
@kkrgwbj would you be able to submit me a Gist snippet - or even a Git project - where I can reproduce the benchmark tests you did?

**Comment by stevenschlansker** [comment]
I did some performance work, and my initial conclusion was that yes we can make some concrete improvements in jdbi, but the synthetic benchmark wasn't a good proxy for "real" performance.  We can and should improve our microbenchmarks but the problems identified probably wouldn't manifest in a "real" database setup (where network traffic etc is expensive) and can be mitigated by using non-reflection mappers.

**Comment by qualidafial** [comment]
#585 would improve performance for reflective mappers v3, by inspecting the bean and doing column matching only once for the entire result set.

**Comment by qualidafial** [comment]
#585 merged--this should be addressed going forward in v3

**Comment by stevenschlansker** [comment]
Yes, that should fix the obvious cases in the mapping pipeline going forward in v3.  Let's close this issue as it doesn't track any particular performance problem anymore.  But if we find more bottlenecks, please open more issues as they come up :)
